src/daemon/utils.rs

```diff
@@ -0,0 +1,84 @@
+#[cfg(test)]
+pub mod test_utils {
+    use crate::{
+        jsonrpc::UserRole,
+        revaultd::RevaultD,
+        threadmessages::{BitcoindMessageOut, SigFetcherMessageOut},
+        RpcUtils,
+    };
+    use common::config::Config;
+
+    use std::{
+        fs,
+        path::PathBuf,
+        sync::{mpsc, Arc, RwLock},
+        thread,
+    };
+
+    // Create a RevaultD state instance using a scratch data directory, trying to be portable
+    // across UNIX, MacOS, and Windows
+    pub fn dummy_revaultd(datadir_name: &'static str, role: UserRole) -> RevaultD {
+        let repo_root = PathBuf::from(file!())
+            .parent()
+            .unwrap()
+            .parent()
+            .unwrap()
+            .parent()
+            .unwrap()
+            .to_path_buf();
+        let datadir_path: PathBuf = [repo_root.to_str().unwrap(), "test_data", datadir_name]
```

Comment:
I think we should rather use scratch datadir in the cwd as done in the `miradord` pull request.

Reply:
I don't think it's a good idea to use cwd just like that - you're going to have messy, dangling directories if you stop the tests before they finish (for example if you run the wrong command). I refactored it to be similar to the one in miradord, but I use `/tmp/revaultd_test_data/` as the base directory, what do you think?

Reply:
That it's not portable :/ Unfortunately it's messy but really the "less dirty" solution i could find

Reply:
I tend to forget that some people don't use linux 😅 Should be good now